packages/appium/lib/appium.js

```diff
@@ -377,16 +373,14 @@ class AppiumDriver extends DriverCore {
    * @param {import('../types/extension').DriverClass} InnerDriver
    * @returns {Promise<DriverData[]>}}
    */
-  async curSessionDataForDriver (InnerDriver) {
-    const sessions = await sessionsListGuard.acquire(AppiumDriver.name, () => this.sessions);
-    const data = _.compact(_.values(sessions)
-                   .filter((s) => s.constructor.name === InnerDriver.name)
-                   .map((s) => s.driverData));
-    for (let datum of data) {
+  async curSessionDataForDriver (InnerDriver) { // eslint-disable-line require-await
+    const data = _.compact(_.values(this.sessions)
+      .filter((s) => s.constructor.name === InnerDriver.name)
```

Comment:
1. why not use `s instanceof InnerDriver` instead of `s.constructor.name` comparison?
2. we can remove the `_.compact` (I may have even added this?) and change the `filter` callback instead:

```js
// untested
.filter((s) => s.constructor.name === InnerDriver.name && s.driverData)
```

Reply:
I remember there were some issues with instanceof call if a class name was imported from a different package. Maybe @jlipps knows more about it

> we can remove the _.compact (I may have even added this?) and change the filter callback instead:

I don't want to change too much here, since this is not the intention of that PR

Reply:
yeah `instanceof` checks don't work for classes that come from two different versions of the same package. not sure if that is a possibility here but it's something we ran into a lot in earlier versions of appium (we couldn't do `instanceof` checks on appium errors because we might be dealing with different versions of BaseDriver imported by a driver vs imported by Appium)